### PR TITLE
feat: enable nvme support for rpi_generic

### PR DESCRIPTION
**Pull request by xvzf**
> ~Note: Boot offset is still wrong; still checking; manually booting works~
>
> Edit: works fine now 🎉 

This allows the RaspberryPi CM4 to use an M.2 NVMe SSD as boot drive.
It also contains a fix for https://github.com/siderolabs/pkgs/issues/658

Fixes https://github.com/siderolabs/talos/issues/5912

Signed-off-by: Matthias Riegler <me@xvzf.tech>

### CODE REVIEW

File: u-boot/rpi_arm64/0001-nvme-support.patch [hunk 1]
```diff
@@ -0,0 +1,407 @@
+# This patch is based on https://github.com/home-assistant/operating-system/commits/755239536812b60f7ebe1c4c679b39597c3f560d/buildroot-external/board/raspberrypi/patches
```
Comment (frezbo) [inline u-boot/rpi_arm64/0001-nvme-support.patch:1]: let's use the individual patches as is from buildroot-external, as opposed to a single file, makes it easier to remove patches when things gets upstreamed

Comment (xvzf) [reply u-boot/rpi_arm64/0001-nvme-support.patch:1]: I had to modify some patches to work with the 2022.07. release & therefore bundled them. I'll update this PR, giving u-boot 2023.01 (released yesterday) a spin to see if it works for the remaining platforms (jetson&rockpi)

Comment (xvzf) [reply u-boot/rpi_arm64/0001-nvme-support.patch:1]: Didn't work out for the jetson

Comment (frezbo) [reply u-boot/rpi_arm64/0001-nvme-support.patch]: yeh, the jetson is one thing preventing us from upgrading right now :cry:

**Comment by xvzf** [comment]
so this is not ready for merging yet, either I have an issue in my compile toolchain or it's not working with the patches and I missed something, will get back here!

File: u-boot/rpi_arm64/0001-nvme-support.patch (hunk 1 above)
Comment (frezbo) [reply u-boot/rpi_arm64/0001-nvme-support.patch]: hmm I seem to have fixed the patches for u-boot 2023.01, might make your work easier

Comment (frezbo) [reply u-boot/rpi_arm64/0001-nvme-support.patch]: first need to test the build :crossed_fingers:

Comment (xvzf) [reply u-boot/rpi_arm64/0001-nvme-support.patch]: Let me know!

I took the 2023.01 patches and backported them to 2022.07

I know took the 2022.01. patches and adapted them for 2022.07 but it's not working - I'll spend some more time on that this evening or on the weekend

Comment (xvzf) [reply u-boot/rpi_arm64/0001-nvme-support.patch]: It did!

**Comment by frezbo** [comment]
@xvzf is this good to go?

**Comment by xvzf** [comment]
@frezbo yep, tested & working

**Review by frezbo — APPROVED** [review]

**Pull request merged**